src/app/client-preview/page.tsx

```diff
@@ -5,32 +5,41 @@ import { apiUrl } from '@/config'
 
 export const revalidate = 0
 
-async function getSettings(): Promise<ISettings> {
-  const { data } = await fetch(`${apiUrl}/api/settings`).then((res) =>
-    res.json(),
+async function getSettings(token: string) {
+  const { data } = await fetch(`${apiUrl}/api/settings?token=${token}`).then(
+    (res) => res.json(),
   )
   return data
 }
 
-async function getClient(clientId: string): Promise<IClient> {
-  const res = await fetch(`${apiUrl}/api/client?clientId=${clientId}`)
+async function getClient(clientId: string, token: string): Promise<IClient> {
+  const res = await fetch(
+    `${apiUrl}/api/client?clientId=${clientId}&token=${token}`,
+  )
   const { data } = await res.json()
   return data
 }
 
-async function getCompany(companyId: string) {
-  const res = await fetch(`${apiUrl}/api/companies?companyId=${companyId}`)
+async function getCompany(companyId: string, token: string) {
+  const res = await fetch(
+    `${apiUrl}/api/companies?companyId=${companyId}&token=${token}`,
+  )
   const { data } = await res.json()
   return data
 }
 
-export default async function ClientPreviewPage() {
-  const settings = await getSettings()
+export default async function ClientPreviewPage({
+  searchParams,
+}: {
+  searchParams: { token: string }
+}) {
+  const { token } = searchParams
+  const settings = await getSettings(token)
 
   //the clientId will be dynamic later
-  const _client = await getClient('a2b66ac1-eedb-4b1c-8b6e-941b71583f3f')
+  const _client = await getClient('80ca5645-73e8-4a22-9bf5-cc5327355972', token)
```

Comment:
we should call .me endpoint to get current client user and use the id from that

Reply:
@sajjanstha I think we should add another API for this.

Reply:
integrated api/me api to the preview route